### PR TITLE
[bitnami/argo-cd] Allowing multi namespace tendancy

### LINKED ISSUE #27108: [bitnami/argo-cd] ArgoCD repo-server NetworkPolicy uses ApplicationSet values
### Name and Version

bitnami/argo-cd 6.4.6

### What architecture are you using?

amd64

### What steps will reproduce the bug?

1. Install the chart with `repoServer.networkPolicy.enabled: true`

### Are you using any custom parameters or values?

`repoServer.networkPolicy.enabled: true`

### What is the expected behavior?

The `NetworkPolicy` resource generated for the repo-server component should have the values specified in the `repoServer.networkPolicy` section.

### What do you see instead?

It has the values from the `applicationSet.networkPolicy` section instead.

### Additional information

See [templates/repo-server/networkpolicy.yaml in the current latest commit](https://github.com/bitnami/charts/blob/dd705237d4f14d7005b707075995185272fb4882/bitnami/argo-cd/templates/repo-server/networkpolicy.yaml) - all the templates use `.Values.applicationSet` instead of `.Values.repoServer`.

### LINKED ISSUE #27449: [bitnami/argo-cd] Argo-CD Cluster Roles to support multiple namespaces, Dynamic Cluster Distribution issue with controller.
### Name and Version

bitnami/argo-cd 6.4.8

### What is the problem this feature will solve?

This will bring the bitnami helm chart for argo-cd to support multiple namespaces by providing cluster roles for services that need them and allow configuration from helm user.

## Adding cluster roles for the following:

```yaml
applicationSet:
  # Default value false
  clusterAdminAccess: false
  clusterRoleRules: []

notifications:
  # Default value false
  clusterAdminAccess: false
  clusterRoleRules: []

repoServer:
  # Default value false
  clusterAdminAccess: false
  clusterRoleRules: []

server:
  clusterAdminAccess: true
  clusterRoleRules: []
```

### Upstream examples (Check Mark = PR work done):
- [x] [applicationset](https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-applicationset/clusterrole.yaml)
- [x] [notifications](https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-notifications/clusterrole.yaml)
- [x] [repoServer](https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-repo-server/clusterrole.yaml)
- [x] [server](https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-server/clusterrole.yaml)
  - This needs to be updated this currently has a cluster role, however there's no way to disable this access and given by default. Default is to give cluster admin access which should not be the case unless explicitly allowed.

### What:
When enabling [Dynamic Cluster Distribution](https://argo-cd.readthedocs.io/en/stable/operator-manual/dynamic-cluster-distribution/#enabling-dynamic-distribution-of-clusters) env on the controller it causes a issue with the application-set not being able to find the deployment name for the app-controller when multi-namespace is used

By default argo-cd looks for argocd-application-controller, this causes the applicationset to fail unless the provided env is set on the controller [ARGOCD_APPLICATION_CONTROLLER_NAME](https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-application-controller/deployment.yaml#L83)

### Changes:
Adding the ARGOCD_APPLICATION_CONTROLLER_NAME as a default env to be made providing the templated name to the controller in the templates [here](https://github.com/bitnami/charts/blob/main/bitnami/argo-cd/templates/application-controller/deployment.yaml#L166)

## Additionally fix:

PR:
- #27108

## Documentaion for setting up any-namespace
New documentation in read-me for setting up any-namespace configurtation, just explaining what values to set to allow this. Links to upstream will be provided in this documentation.

### Upstream documentation:
- [Applications in any namespace](https://argo-cd.readthedocs.io/en/stable/operator-manual/app-any-namespace/)
- [ApplicationSet in any namespace](https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Appset-Any-Namespace/)
- [Notification in any namespace](https://argo-cd.readthedocs.io/en/stable/operator-manual/notifications/#namespace-based-configuration)
- [Reconfigure Argo CD to allow certain namespaces](https://argo-cd.readthedocs.io/en/stable/operator-manual/app-any-namespace/#reconfigure-argo-cd-to-allow-certain-namespaces)
- [Dynamic Cluster Distribution](https://argo-cd.readthedocs.io/en/stable/operator-manual/dynamic-cluster-distribution/)

## Using extraDeploy to provide the cluster roles and bindings in the values file, and controller.extraEnvVars

```yaml
# This requires makes for a unessaryly long values file
# Also, requires you know the templated service account name for each service inside the cluster rolebinding
extraDeploy:
- CLUSTER-ROLE-FOR-applicationset
- CLUSTER-ROLEBINDING-FOR-applicationset
- CLUSTER-ROLE-FOR-notifications
- CLUSTER-ROLEBINDING-FOR-notifications
- CLUSTER-ROLE-FOR-repoServer
- CLUSTER-ROLEBINDING-FOR-repoServer
- CLUSTER-ROLE-FOR-server
- CLUSTER-ROLEBINDING-FOR-server

controller:
  extraEnvVars:
  - name: ARGOCD_APPLICATION_CONTROLLER_NAME
    value: MANUALLY-SETTING-NAME-OF-TEMPLATED-DEPLOYMENT-NAME
    
```

### PR DESCRIPTION
**Pull request by Jeremy-Boyle**
### Description of the change

This will bring the bitnami helm chart for argo-cd to support multiple namespaces by providing cluster roles for services that need them and allow configuration from helm user.

Updates read-me to provide documentation on enabling muli-namespace configuration referencing argo-cd docs. 

### Benefits

Fixes the templating issue in #27108, allows clusterRole to be provided from chart to allow multiple namespace see issue #27449

### Possible drawbacks

Potentially need to ensure that upstream doesn't add / change additional requirements for cluster role, clusterRoleRules provided to mitigate any issues requiring a immediate fix from bitnami team. 

### Applicable issues

- fixes #27449
- fixes #27108

### Additional information

See issue #27449 for more information

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/argo-cd] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by Jeremy-Boyle** [comment]
# Testing values file with just clusterAdminAccess.
```yaml
server:
  clusterAdminAccess: true

repoServer:
  clusterAdminAccess: true

notifications:
  enabled: true
  clusterAdminAccess: true

applicationSet:
  enabled: true
  clusterAdminAccess: true
```

## output
```yaml
# Source: argo-cd/templates/applicationset/clusterrole.yaml
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: argocd-argo-cd-applicationset-controller
  namespace: "argocd"
  labels:
    app.kubernetes.io/instance: argocd
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: argo-cd
    app.kubernetes.io/version: 2.11.3
    helm.sh/chart: argo-cd-6.4.9
rules:
  - apiGroups:
    - argoproj.io
    resources:
    - applications
    - applicationsets
    - applicationsets/finalizers
    - applicationsets/status
    verbs:
    - create
    - delete
    - get
    - list
    - patch
    - update
    - watch
  - apiGroups:
    - argoproj.io
    resources:
    - applicationsets/status
    verbs:
    - get
    - patch
    - update
  - apiGroups:
    - argoproj.io
    resources:
    - appprojects
    verbs:
    - get
  - apiGroups:
    - ""
    resources:
    - events
    verbs:
    - create
    - get
    - list
    - patch
    - watch
  - apiGroups:
    - ""
    resources:
    - secrets
    - configmaps
    verbs:
    - get
    - list
    - watch
  - apiGroups:
    - apps
    - extensions
    resources:
    - deployments
    verbs:
    - get
    - list
    - watch
  # Leader election
  - apiGroups:
    - ""
    resources:
    - configmaps
    verbs:
    - create
    - update
    - delete
    - patch
  - apiGroups:
    - coordination.k8s.io
    resources:
    - leases
    verbs:
    - create
    - delete
    - get
    - list
    - patch
    - update
    - watch
---
# Source: argo-cd/templates/notifications/clusterrole.yaml
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: argocd-argo-cd-notifications
  namespace: "argocd"
  labels:
    app.kubernetes.io/instance: argocd
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: argo-cd
    app.kubernetes.io/version: 2.11.3
    helm.sh/chart: argo-cd-6.4.9
rules:
  - apiGroups:
    - argoproj.io
    resources:
    - applications
    - appprojects
    verbs:
    - get
    - list
    - watch
    - update
    - patch
  - apiGroups:
    - ""
    resources:
    - configmaps
    - secrets
    verbs:
    - list
    - watch
    # Needed for multi-tenancy namespace
    - create
  - apiGroups:
    - ""
    resourceNames:
    - argocd-notifications-cm
    resources:
    - configmaps
    verbs:
    - get
  - apiGroups:
    - ""
    resourceNames:
    - argocd-notifications-secret
    resources:
    - secrets
    verbs:
    - get
---
# Source: argo-cd/templates/repo-server/clusterrole.yaml
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: argocd-argo-cd-repo-server
  namespace: "argocd"
  labels:
    app.kubernetes.io/instance: argocd
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: argo-cd
    app.kubernetes.io/version: 2.11.3
    helm.sh/chart: argo-cd-6.4.9
rules:
  - apiGroups:
      - argoproj.io
    resources:
      - applications
      - applicationset
    verbs:
      - get
      - list
      - watch
---
# Source: argo-cd/templates/server/clusterrole.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: argocd-argo-cd-argocd-server
  labels:
    app.kubernetes.io/instance: argocd
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: argo-cd
    app.kubernetes.io/version: 2.11.3
    helm.sh/chart: argo-cd-6.4.9
rules:
  - apiGroups:
      - '*'
    resources:
      - '*'
    verbs:
      - delete
      - get
      - patch
  - apiGroups:
      - ""
    resources:
      - events
    verbs:
      - list
      # Needed for multi-tenancy namespace
      - create
  - apiGroups:
      - ""
    resources:
      - pods
      - pods/log
    verbs:
      - get
  - apiGroups:
      - argoproj.io
    resources:
      - applications
    verbs:
      - get
      - list
      - update
      - watch
      # Needed for multi-tenancy namespace
      - create
      - delete
      - patch
---
# Source: argo-cd/templates/application-controller/clusterrolebinding.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: argocd-argo-cd-argocd-app-controller
  labels:
    app.kubernetes.io/instance: argocd
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: argo-cd
    app.kubernetes.io/version: 2.11.3
    helm.sh/chart: argo-cd-6.4.9
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: argocd-argo-cd-argocd-app-controller
subjects:
  - kind: ServiceAccount
    name: argocd-argo-cd-argocd-app-controller
    namespace: argocd
---
# Source: argo-cd/templates/applicationset/clusterrolebinding.yaml
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: argocd-argo-cd-applicationset-controller
  namespace: "argocd"
  labels:
    app.kubernetes.io/instance: argocd
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: argo-cd
    app.kubernetes.io/version: 2.11.3
    helm.sh/chart: argo-cd-6.4.9
subjects:
  - kind: ServiceAccount
    name: argocd-argo-cd-applicationset-controller
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: argocd-argo-cd-applicationset-controller
---
# Source: argo-cd/templates/notifications/clusterrolebinding.yaml
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: argocd-argo-cd-notifications
  namespace: "argocd"
  labels:
    app.kubernetes.io/instance: argocd
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: argo-cd
    app.kubernetes.io/version: 2.11.3
    helm.sh/chart: argo-cd-6.4.9
subjects:
  - kind: ServiceAccount
    name: argocd-argo-cd-notifications
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: argocd-argo-cd-notifications
---
# Source: argo-cd/templates/repo-server/clusterrolebinding.yaml
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: argocd-argo-cd-repo-server
  namespace: "argocd"
  labels:
    app.kubernetes.io/instance: argocd
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: argo-cd
    app.kubernetes.io/version: 2.11.3
    helm.sh/chart: argo-cd-6.4.9
subjects:
  - kind: ServiceAccount
    name: argocd-argo-cd-argocd-repo-server
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: argocd-argo-cd-repo-server
---
# Source: argo-cd/templates/server/clusterrolebinding.yaml
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: argocd-argo-cd-argocd-server
  labels:
    app.kubernetes.io/instance: argocd
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: argo-cd
    app.kubernetes.io/version: 2.11.3
    helm.sh/chart: argo-cd-6.4.9
    app.kubernetes.io/component: server
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: argocd-argo-cd-argocd-server
subjects:
  - kind: ServiceAccount
    name: argocd-argo-cd-argocd-server
    namespace: argocd
```

**Comment by Jeremy-Boyle** [comment]
# Controller env, being set now

```yaml
env:
- name: ARGOCD_APPLICATION_CONTROLLER_NAME
  value: argocd-argo-cd-app-controller
```

**Comment by Jeremy-Boyle** [comment]
PR, completed, please review, and suggest changes !

Thank you !

**Comment by Jeremy-Boyle** [comment]
Fixed, failing jobs due to linting.

File: bitnami/argo-cd/Chart.yaml [hunk 1]
```diff
@@ -39,4 +39,4 @@ maintainers:
 name: argo-cd
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/argo-cd
-version: 6.4.8
+version: 6.4.9
```
Comment (jotamartos) [inline bitnami/argo-cd/Chart.yaml:42]: As you are adding new parameters, we should release a new minor version 

```suggestion
version: 6.5.0
```

**Review by jotamartos — COMMENTED** [review]
Changes LGTM! I'll apply the suggestion and merge the changes once the tests succeed

**Review by jotamartos — APPROVED** [review]

**Pull request merged**